cpuinfo.go

```diff
@@ -183,22 +172,16 @@ func parseCPUInfoX86(info []byte) ([]CPUInfo, error) {
 			cpuinfo[i].PowerManagement = field[1]
 		}
 	}
-	return cpuinfo, nil
+	return cpuinfo, scanner.Err()
```

Comment:
If `scanner.Err()` is not nil here, I'm wondering if we should return a nil instead of cpuinfo.  At least that's the pattern we've usually followed for other parts of this library.

Reply:
I think this is fine.

Reply:
> if scanner.Err() is not nil here, I'm wondering if we should return a nil instead of cpuinfo. At least that's the pattern we've usually followed for other parts of this library.

It seems like an unnecessary complication. A user should always check for the error first. If the error is non-nil, there are no guarantees wrt what the result would be.

Reply:
Ok makes sense, I'm fine with this as-is.